src/Microsoft.AspNetCore.Sockets.Client/SendMessage.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Sockets.Client
+{
+    public struct SendMessage
+    {
+        public MessageType Type { get; }
+        public byte[] Payload { get; }
+        public TaskCompletionSource<object> SendResult { get; }
```

Comment:
We use the channel to send the user request from the application to the transport. Data sent from the transport to the connection are just messages received by the transport from the server. The connection has a loop running on a separate thread that removes these messages from the channel and raise the Received event. The issue was that before this change `SendAsync` would complete once the message was queued to the channel and we were not able to tell the user if sending the message actually succeeded. They would get the exception in the `Closed` event but in general it was impossible to correlate this exception with the actual `SendAsync` call that failed. Now the `SendMessage` contains the tcs which is used by the transport let the user know if sending the message succeeded. 

(EndOfMessage is not relevant here - it is just an artifact of sharing code with the server - this is for partial websocket messages which we don't allow on the client. The WebSocketsTransport on the client will buffer until the entire message is received and only then will push the message to the channel,)